### PR TITLE
PortalSettings resource added to the Azure APIM specification

### PR DESCRIPTION
**Pull request by promoisha**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [+ ] The title of the PR is clear and informative.
- [+ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [+ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- +[x] If applicable, the PR references the bug/issue that it fixes.
- [+ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [+ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- +[x] My spec meets the review criteria:
  - [+ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [+ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ +] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimportalsettings.json [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "DelegationSettings"
+        ],
+        "operationId": "DelegationSettings_CreateOrUpdate",
```
Comment (solankisamir) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimportalsettings.json]: Add this file to the readme.md under 2017-03-01

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/examples/ApiManagementPortalSettingsUpdateDelegation.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "serviceName": "apimService1",
+    "resourceGroupName": "rg1",
+    "api-version": "2017-03-01",
+    "subscriptionId": "subid",
+    "uid": "5931a75ae4bbd512288c680b",
+    "If-Match": "*",
+    "parameters": {
+      "properties": {
+        "url": "http://contoso.com/delegation",
+        "validationKey": "nVF7aKIvr9mV/RM5lOD0sYoi8ThXTRHQP7o66hvUmjCDkPKR3qxPu/otJcNciz2aQdqPuzJH3ECG4TU2yZjQ7Q==",
```
Comment (alvadb) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/examples/ApiManagementPortalSettingsUpdateDelegation.json]: Should this key be public?

**Review by alvadb — CHANGES_REQUESTED** [review]

**Comment by alvadb** [comment]
There are some linter errors (https://travis-ci.org/Azure/azure-rest-api-specs/jobs/275169528). Please fix.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/examples/ApiManagementPortalSettingsUpdateDelegation.json (hunk 2 above)
Comment (promoisha) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/examples/ApiManagementPortalSettingsUpdateDelegation.json]: Yes, as it's not the actual key, but some hashed value of the URL.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimportalsettings.json (hunk 1 above)
Comment (promoisha) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimportalsettings.json]: Done.

**Comment by promoisha** [comment]
@alvadb I fixed a bunch of linter warnings and left some minor warnings/errors out of the fix. The error that  'Operations API must be implemented' is not relevant to the resource I'm adding because it's not going to be registered in ARM.

**Comment by alvadb** [comment]
Most of the errors from the linter appear to be from missing examples but there is one that is a casing issue: 
{
  "type": "Error",
  "code": "DefinitionsPropertiesNamesCamelCase",
  "message": "Property named: 'Value', for definition: 'QuotaCounterContract' must follow camelCase style. Example: 'value'.",
  "id": "R3016",
  "validationCategory": "RPCViolation",
  "providerNamespace": null,
  "resourceType": null,
  "sources": [
    "file:///home/travis/build/Azure/azure-rest-api-specs/specification/apimanagement/resource-manager/Microsoft.ApiManagement/2017-03-01/apimquotas.json:284:8 ($.definitions.QuotaCounterContract.properties.Value)"
  ]
}

Can you fix?

Also, there are a couple of semantic issues and the errors appear to indicate that there are duplicate paths.  Please check: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/276094929.

**Comment by promoisha** [comment]
@alvadb We'll create a separate task for the error you mentioned, just to keep the current PR clean and to not block it with unrelated fixes. Regarding semantics, waiting for the response from @veronicagg.

**Comment by veronicagg** [comment]
@promoisha I've replied on the related issue, thanks.

**Comment by promoisha** [comment]
@alvadb I changed the code in order to put the enum values directly in the path, as per [https://github.com/Azure/azure-rest-api-specs/issues/1678#issuecomment-331509672](url). There should be no issues with the portalsettings spec by now. Thanks.

**Comment by alvadb** [comment]
Will merge with the linter errors since they will be address in a subsequent check in.

**Review by alvadb — APPROVED** [review]

**Pull request merged**